inmoose/consensus_clustering/consensus_clustering.py

```diff
@@ -0,0 +1,360 @@
+        """
+        assert self.consensus_matrices is not None, "First compute consensus clustering"
+        return np.array([(Ab-Aa)/Aa if i > 2 else Aa/10
+                                    for Ab, Aa, i in zip(self.Ak[1:], self.Ak[:-1], range(self.min_k, self.max_k))])
+        
+    def compute_area_under_curve(self):
+        """
+        Compute area under the CDFs curve 
+
+        Returns
+        -----------
+        area_under_curve
+            array of the area under the CDF for each cluster number
+        """
+        assert self.consensus_matrices is not None, "First compute consensus clustering"
+        area_under_curve = np.zeros(self.max_k-self.min_k+1)
+        for i, m in enumerate(self.consensus_matrices):
+            hist, bins = np.histogram(m.ravel(), density=True)
+            area_under_curve[i] = np.sum(h*(b-a)
+                             for b, a, h in zip(bins[1:], bins[:-1], np.cumsum(hist)))
+        return area_under_curve
+    
+    def compute_summary_statistics(self, k):
+        """
+        For one prediction, compute a summary statistics, cluster consensus and item consensus, showing cluster stability and most representative cluster items.
+        
+        Cluster consensus is defined as the average consensus index between all pairs of items belonging to the cluster.
+        Item consensus is defined as the average consensus index between item ei and all the (other) items in a cluster. 
+
+
+        Arguments
+        -----------
+        k: int
+            Number of clusters
+        Returns
+        -----------
+        predictions
+            Array of the predicted cluster 
+        clusters_consensus
+            Array of cluster consensus
+        items_consensus
+            Array of the item consensus (nb_items * k)
+
+        """
+        assert self.consensus_matrices is not None, "First compute consensus clustering"
+        predictions = self.predict(k)
+        clusters_consensus = self.compute_clusters_consensus(predictions, k)
+        items_consensus = self.compute_items_consensus(predictions, k)
+        return predictions, clusters_consensus, items_consensus
+    
+    def compute_clusters_consensus(self, prediction, k):
+        """
+        For one prediction, compute clusters consensus, showing cluster stability.
+        
+        Cluster consensus is defined as the average consensus index between all pairs of items belonging to the cluster.
+        Arguments
+        -----------
+        prediction: nparray
+            Array of the predicted cluster 
+        k: int
+            Number of clusters
+        Returns
+        -----------
+        clusters_consensus
+            Array of cluster consensus
+        """
+        clusters_consensus = np.zeros(k)
+        for clust in range(k):
+            clust_size = len(np.where(prediction==clust)[0])
+            ids_ = np.array(list(combinations(np.sort(np.where(prediction==0)[0]), 2))).T
```

Comment:
prediction == clust here?
Maybe you can compute this once for each cluster instead of twice?